ddtrace/propagation/http.py

```diff
@@ -859,7 +866,15 @@ def my_controller(url, headers):
 
             # loop through the extract propagation styles specified in order
             for prop_style in config._propagation_style_extract:
-                propagator = _PROP_STYLES[prop_style]
+                if (
```

Comment:
google cloud function traces will have both the `x-cloud-trace-context` and `traceparent` headers set. If the cloud function was not called by an upstream datadog instrumented service (meaning no `x-datadog-trace-id` is set), we want to enter the regular traceparent extractor (where the parent_id will be intentionally left empty).

Otherwise (datadog trace context headers are set, but `traceparent is also set`), use the datadog extractor.